{
  if (doFail.get() && count++ % 2 == 1) {
    throw new IOException("Simulated network outage");
  }
}
